File: quan.cs/Test/Config/Common/ErrorCodeConfig.cs
using System.Collections.Generic;
using System.Collections.Immutable;
using Newtonsoft.Json.Linq;
using Quan.Common;
using Quan.Config;

namespace Test.Config.Common
{
    /// <summary>
	/// 错误码<br/>
	/// 自动生成
	/// </summary>
    public class ErrorCodeConfig : ConfigBase
    {
        /// <summary>
        /// ID
        /// </summary>
        public readonly int Id;

        /// <summary>
        /// Key
        /// </summary>
        public readonly string Key;

        /// <summary>
        /// 文本
        /// </summary>
        public readonly string Text;


        public ErrorCodeConfig(JObject json) : base(json)
        {
            Id = json["id"]?.Value<int>() ?? default;
            Key = json["key"]?.Value<string>() ?? "";
            Text = json["text"]?.Value<string>() ?? "";
        }

        protected override ConfigBase Create(JObject json)
        {
            return new ErrorCodeConfig(json);
        }


        public override string ToString()
        {
            return "ErrorCodeConfig{" +
                   "Id=" + Id.ToString2() +
                   ",Key='" + Key + '\'' +
                   ",Text='" + Text + '\'' +
                   '}';
        }


        // 所有ErrorCodeConfig
        private static volatile IList<ErrorCodeConfig> _configs = new List<ErrorCodeConfig>();

        // ID
        private static volatile IDictionary<int, ErrorCodeConfig> _idConfigs = new Dictionary<int, ErrorCodeConfig>();

        // Key
        private static volatile IDictionary<string, ErrorCodeConfig> _keyConfigs = new Dictionary<string, ErrorCodeConfig>();

        public static IList<ErrorCodeConfig> GetConfigs()
        {
            return _configs;
        }

        public static IDictionary<int, ErrorCodeConfig> GetIdConfigs()
        {
            return _idConfigs;
        }

        public static ErrorCodeConfig GetById(int id)
        {
            _idConfigs.TryGetValue(id, out var result);
            return result;
        }

        public static IDictionary<string, ErrorCodeConfig> GetKeyConfigs()
        {
            return _keyConfigs;
        }

        public static ErrorCodeConfig GetByKey(string key)
        {
            _keyConfigs.TryGetValue(key, out var result);
            return result;
        }


        public static void Load(IList<ErrorCodeConfig> configs)
        {
            IDictionary<int, ErrorCodeConfig> idConfigs = new Dictionary<int, ErrorCodeConfig>();
            IDictionary<string, ErrorCodeConfig> keyConfigs = new Dictionary<string, ErrorCodeConfig>();

            foreach (var config in configs)
            {
                Load(idConfigs, config, config.Id);
                Load(keyConfigs, config, config.Key);
            }

            configs = configs.ToImmutableList();
            idConfigs = ToImmutableDictionary(idConfigs);
            keyConfigs = ToImmutableDictionary(keyConfigs);

            _configs = configs;
            _idConfigs = idConfigs;
            _keyConfigs = keyConfigs;
        }
    }
}

File: quan.java/config/src/test/java/quan/config/common/ErrorCodeConfig.java
package quan.config.common;

import java.util.*;
import com.alibaba.fastjson.*;
import quan.config.*;

/**
* 错误码<br/>
* 自动生成
*/
public class ErrorCodeConfig extends Config {

    //ID
    protected final int id;

    //Key
    protected final String key;

    //文本
    protected final String text;


    public ErrorCodeConfig(JSONObject json) {
        super(json);

        this.id = json.getIntValue("id");
        this.key = json.getOrDefault("key", "").toString();
        this.text = json.getOrDefault("text", "").toString();
    }

    /**
     * ID
     */
    public final int getId() {
        return id;
    }

    /**
     * Key
     */
    public final String getKey() {
        return key;
    }

    /**
     * 文本
     */
    public final String getText() {
        return text;
    }


    @Override
    protected ErrorCodeConfig create(JSONObject json) {
        return new ErrorCodeConfig(json);
    }

    @Override
    public String toString() {
        return "ErrorCodeConfig{" +
                "id=" + id +
                ",key='" + key + '\'' +
                ",text='" + text + '\'' +
                '}';

    }


    // 所有ErrorCodeConfig
    private static volatile List<ErrorCodeConfig> configs = new ArrayList<>();

    //ID
    private static volatile Map<Integer, ErrorCodeConfig> idConfigs = new HashMap<>();

    //Key
    private static volatile Map<String, ErrorCodeConfig> keyConfigs = new HashMap<>();

    public static List<ErrorCodeConfig> getConfigs() {
        return configs;
    }

    public static Map<Integer, ErrorCodeConfig> getIdConfigs() {
        return idConfigs;
    }

    public static ErrorCodeConfig getById(int id) {
        return idConfigs.get(id);
    }

    public static Map<String, ErrorCodeConfig> getKeyConfigs() {
        return keyConfigs;
    }

    public static ErrorCodeConfig getByKey(String key) {
        return keyConfigs.get(key);
    }


    @SuppressWarnings({"unchecked"})
    public static List<String> load(List<ErrorCodeConfig> configs) {
        Map<Integer, ErrorCodeConfig> idConfigs = new HashMap<>();
        Map<String, ErrorCodeConfig> keyConfigs = new HashMap<>();

        List<String> errors = new ArrayList<>();

        for (ErrorCodeConfig config : configs) {
            load(idConfigs, errors, config, true, Arrays.asList("id"), config.id);
            load(keyConfigs, errors, config, true, Arrays.asList("key"), config.key);
        }

        configs = Collections.unmodifiableList(configs);
        idConfigs = unmodifiableMap(idConfigs);
        keyConfigs = unmodifiableMap(keyConfigs);

        ErrorCodeConfig.configs = configs;
        ErrorCodeConfig.idConfigs = idConfigs;
        ErrorCodeConfig.keyConfigs = keyConfigs;

        return errors;
    }

}
